martin/src/config/file/main.rs

```diff
@@ -91,33 +91,88 @@ impl Config {
             self.srv.base_path = Some(parse_base_path(path)?);
         }
         #[cfg(feature = "postgres")]
-        let pg_prefix = if matches!(self.postgres, OptOneMany::One(_)) {
-            "postgres."
-        } else {
-            "postgres[]."
-        };
-        #[cfg(feature = "postgres")]
-        for pg in self.postgres.iter_mut() {
-            res.extend(pg.finalize(pg_prefix)?);
+        {
+            let pg_prefix = if matches!(self.postgres, OptOneMany::One(_)) {
+                "postgres"
+            } else {
+                "postgres[]"
+            };
+            for pg in self.postgres.iter_mut() {
+                pg.finalize()?;
+                res.extend(
+                    pg.get_unrecognized_keys()
```

Comment:
can we have a helper function for this? since `get_unrecognized_keys` is part of the `ConfigurationLivecycleHooks`, we can simply add a new function to `ConfigurationLivecycleHooks` together with a simple implementation